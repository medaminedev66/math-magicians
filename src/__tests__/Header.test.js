import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';

describe('test the Quote components', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const root = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Header />
      </Router>,
      root,
    );
    expect(root.querySelector('h1').textContent).toBe('Math Magicians');
  });
});
