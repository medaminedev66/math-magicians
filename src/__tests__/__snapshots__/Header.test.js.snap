// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the Quote components matches the snapshot 1`] = `
<div
  className="header"
>
  <div
    className="logo"
  >
    <h1>
      Math Magicians
    </h1>
  </div>
  <nav>
    <ul
      className="nav"
    >
      <li
        data-testid={1}
      >
        <a
          aria-current="page"
          className="active-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          home
        </a>
      </li>
      <li
        data-testid={2}
      >
        <a
          aria-current={null}
          className="active-link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        data-testid={3}
      >
        <a
          aria-current={null}
          className="active-link"
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
